plugins/inputs/zfs/zfs_linux.go

```diff
@@ -45,36 +80,99 @@ func getTags(pools []poolInfo) map[string]string {
 	return map[string]string{"pools": poolNames}
 }
 
-func gatherPoolStats(pool poolInfo, acc telegraf.Accumulator) error {
-	lines, err := internal.ReadLines(pool.ioFilename)
-	if err != nil {
-		return err
-	}
-
-	if len(lines) != 3 {
-		return err
+func gather(lines []string, fileLines int) ([]string, []string, error) {
+	if len(lines) != fileLines {
+		return nil, nil, errors.New("Expected lines in kstat does not match")
 	}
 
 	keys := strings.Fields(lines[1])
 	values := strings.Fields(lines[2])
+	if len(keys) != len(values) {
+		return nil, nil, fmt.Errorf("Key and value count don't match Keys:%v Values:%v", keys, values)
```

Comment:
Errors should start with lowercase letters in Golang...
```suggestion
		return nil, nil, fmt.Errorf("key and value count don't match Keys:%v Values:%v", keys, values)
```